lib/yeti_logger.rb

```diff
@@ -74,14 +76,21 @@ module YetiLogger
   # passed to the log method.
   # Define these methods explicitly allows the use of yield.
   module LogMethods
+    # See usage at https://github.com/Yesware/yeti_logger/blob/master/README.md#user-based-logging
     def log_debug(obj = nil, ex = nil)
-      if YetiLogger.logger.level <= Logger::DEBUG
+      # In order if increasing likelihood, so it short-circuits as soon as possible
+      should_log_as_info = defined?(Settings) &&
```

Comment:
Since this is a public repo, I think it makes more sense to do this in the `configure` block.  A general way to support this would be to define a proc that returns the array of extra logging user IDs.  Or to be really generic, a proc that decides when to promote a debug message to info.  Something like our implementation could be

```rb
YetiLogger.configure do |config|
  config.promote_debug_to_info? = Proc.new do |log_hash|
    log_hash.with_indifferent_access[:user_id].to_i.in?(Settings.extra_logging_user_ids)
  end
end
```

Then `log_debug` looks something like this:

```rb
  should_log_as_info = YetiLogger.configuration.promote_debug_to_info?.present? &&
    obj.is_a?(Hash) &&
    YetiLogger.configuration.promote_debug_to_info?.call(obj)
```

Reply:
Are you thinking the config block will go in the gem or the apps that use it? I did consider a config block, but if it goes in the app, then we'll have to copypaste identical config blocks into each app. If it stays in the gem, then the gem is just as coupled with `Settings` as in my implementation. Besides, I purposely wrote and tested the logic so the gem still works for people who don't have `Settings`. Finally, the settings-in-redis gem is also open-source, so people do have the option of enabling this.

Reply:
I was thinking it would go in the app.  That way, if someone wants to control the definition of the enabled user IDs without `settings_in_redis`, they can. Maybe they want to hardcode an array of user IDs.  Or maybe they want to key off a different property of the log hash.

As for our internal use of that, we already centralize the configuration of `YetiLogger` in our Rails template and/or other base gems.  We'd only need to update this in one place, and then `bundle update` each app to pick up the changes.

Reply:
Yeah, that make more sense the more I think about it. That way an app that doesn't mind restarting can just stick its user IDs in an evar and not have to spin up a Redis database.

I went with a singleton method instead of a proc so we can have the `?` and don't need the `.call`.

Reply:
Like [this](https://github.com/Yesware/gmail-api-server/commit/23fe7dd3ab97e9d407f6a26123cccd7e734336d8)